neonvm/controllers/vm_qmp_queries.go

```diff
@@ -333,6 +333,65 @@ type QMPRunner interface {
 	Run([]byte) ([]byte, error)
 }
 
+// QmpSetVirtioMem updates virtio-mem to the new target size, returning the previous target.
+//
+// If the new target size is equal to the previous one, this function does nothing but query the
+// target.
+func QmpSetVirtioMem(vm *vmv1.VirtualMachine, targetVirtioMemSize int64) (previous int64, _ error) {
+	// Note: The virtio-mem device only exists when max mem != min mem.
+	// So if min == max, we should just short-cut, skip the queries, and say it's all good.
+	// Refer to the instantiation in neonvm-runner for more.
+	if vm.Spec.Guest.MemorySlots.Min == vm.Spec.Guest.MemorySlots.Max {
+		// if target size is non-zero even though min == max, something went very wrong
+		if targetVirtioMemSize != 0 {
+			panic(fmt.Errorf(
```

Comment:
```suggestion
			panic(fmt.Sprintf(
```

Usually panic's arg is the string.